ChiModules/KEigenvalueSolver/initialize_precursors.cc

```diff
@@ -15,46 +15,45 @@ using namespace LinearBoltzmann;
 /**Computes the point wise delayed neutron precursor concentrations.*/
 void KEigenvalue::Solver::InitializePrecursors()
 {
-  auto pwl = std::static_pointer_cast<SpatialDiscretization_PWLD>(discretization);
-
-  Nj_new_local.assign(Nj_new_local.size(), 0.0);
-
-  // ----- Loop over cells
-  for (auto& cell : grid->local_cells)
-  {
-    // ----- Cell information
-    const auto xs_id = matid_to_xs_map[cell.material_id];
-    auto& xs = material_xs[xs_id];
-    auto cell_fe_view = pwl->GetCellMappingFE(cell.local_id);
-    auto& transport_view = cell_transport_views[cell.local_id];
-
-    // ----- Loop over cell dofs
-    for (int i = 0; i < cell_fe_view->num_nodes; ++i)
-    {
-      size_t ir = transport_view.MapDOF(i,0,0);
-      int64_t jr = pwl->MapDOFLocal(cell, i, Nj_unk_man, 0, 0);
-      double* Nj_newp  = &Nj_new_local[jr];
-      double* phi_newp = &phi_new_local[ir];
-
-      // ----- If a fissial material with precursors
-      if ((xs->is_fissile) and (xs->num_precursors > 0)) {
-        // ----- Loop over precursors
-        for (size_t j = 0; j < xs->num_precursors; ++j)
-        {
-          size_t j_map = precursor_map[xs_id][j];
-
-          // ----- Initialize precursors
-          double coeff = xs->precursor_yield[j] / xs->precursor_lambda[j];
-
-
-          // ----- Loop over groups
-          for (int g = 0; g < groups.size(); ++g)
-            Nj_newp[j_map] += coeff * xs->nu_delayed_sigma_f[g] *
-                              phi_newp[g] / k_eff;
-        }//for j
-      }//if fissile and J > 0
-    } //for i
-  }//for c
+  if (options.use_precursors) {
+    auto pwl = std::static_pointer_cast<SpatialDiscretization_PWLD>(discretization);
+
+    Nj_new_local.assign(Nj_new_local.size(), 0.0);
+
+    // ----- Loop over cells
+    for (auto &cell : grid->local_cells) {
+      // ----- Cell information
+      const auto xs_id = matid_to_xs_map[cell.material_id];
+      auto &xs = material_xs[xs_id];
+      auto cell_fe_view = pwl->GetCellMappingFE(cell.local_id);
+      auto &transport_view = cell_transport_views[cell.local_id];
+
+      // ----- Loop over cell dofs
+      for (int i = 0; i < cell_fe_view->num_nodes; ++i) {
+        size_t ir = transport_view.MapDOF(i, 0, 0);
+        int64_t jr = pwl->MapDOFLocal(cell, i, Nj_unk_man, 0, 0);
```

Comment:
What is Nj_unk_man?
Can you add a comment where that var is defined? If I do not know immediately what it stands for, maybe others in the future will wonder too. Thanks

Reply:
This is the unknown manager for the delayed neutron precursors, named Nj because that is what the delayed precursor vector is named currently. I added a comment in the header file to explain that 'Nj' is the precursors. I just changed 'unk_man' to 'uk_man' to be consistent with unknown manager naming elsewhere. We could change 'Nj_...' to precursor_...' for more readability. Thoughts?

Reply:
I am always for more clarity/readability. I do not think we have have varname with 200 characters but there is likely a good trade-off somewhere.
@Naktakala So uk is like a dof_handler in deal.ii?